src/enums.rs

```diff
@@ -10,6 +11,24 @@ pub enum Error {
     InvalidData,
 }
 
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+        (self as &std::error::Error).description().fmt(f)
+    }
+}
+
+impl std::error::Error for Error {
+    fn description(&self) -> &str {
+        match self {
```

Comment:
Generally we match over `*self` to avoid having to put `&` in front of every pattern matching.